service/storage_proxy.cc

```diff
@@ -3995,6 +4003,7 @@ class abstract_read_executor : public enable_shared_from_this<abstract_read_exec
                 auto&& [result, digests_match] = res.value();
 
                 if (digests_match) {
+                    result->set_last_position(digest_resolver->min_position());
```

Comment:
Can we avoid this if we got a full result (common case)?

Reply:
Yes, in that case the result can keep the last-position it already has. We still have to do the comparison to check that all replicas set the same last-pos.

Reply:
added an if to avoid the copy